### PR TITLE
frontend: created custom dropdown for Country selector

### PR DESCRIPTION
**Pull request by shonsirsha**
To improve our UI & UX, we'd like to have a custom dropdown / select implemented for the Country Selector. This enables search and displays the flag of each region in the option list.

The custom dropdown was implemented using `react-select` pkg.

[Preview]:

https://user-images.githubusercontent.com/28676406/218675245-7375c58b-a74f-4bb6-aadf-0792ca1650ce.mov

### CODE REVIEW

File: frontends/web/src/routes/buy/exchange.tsx [hunk 1]
```diff
@@ -199,23 +189,11 @@ export const Exchange = ({ code, accounts }: TProps) => {
             {regions.length ? (
               <>
                 <div className={style.selectContainer}>
-                  <div>
-                    <SelectedRegionIcon regionCode={selectedRegion.toLowerCase()}/>
-                    <Select
-                      className={style.extraLeftSpace}
```
Comment (thisconnect) [inline frontends/web/src/routes/buy/exchange.tsx:205]: if `extraLeftSpace` isnt' needed anymore please remove from the CSS module.

File: frontends/web/src/routes/buy/exchange.tsx [hunk 2]
```diff
@@ -84,9 +73,9 @@ export const Exchange = ({ code, accounts }: TProps) => {
       return;
     }
     const regionNames = new Intl.DisplayNames([i18n.language], { type: 'region' });
-    const regions = regionList.regions.map(region => ({ value: region.code, text: regionNames.of(region.code) } as TOption));
+    const regions = regionList.regions.map(region => ({ value: region.code, label: regionNames.of(region.code) } as TOption));
```
[the remark this answers is not in the document]
Comment (shonsirsha) [reply frontends/web/src/routes/buy/exchange.tsx:76]: I get what you mean, but I don't think removing the type casting `TOption` is the correct way to do that. I assume you meant something like this:

in
```js
 const regionNames = new Intl.DisplayNames([i18n.language], { type: 'region' });
 const regions = regionList.regions.map(region => ({ value: region.code, label: regionNames.of(region.code) } as TOption));
```
change `regionNames` to
```js
const regionNames = new Intl.DisplayNames([i18n.language], { type: 'region' }) || '';
``` 
? 🤔 


I assume `regionNames` may return undefined, so we could fallback to empty string (if it's not doing that already).

Anyway, `TOption` is needed to make TS happy. `TOption` is just `label: string; value: string;` - this is the type needed to display the select option, regardless if it's empty string or not.

File: frontends/web/package.json [hunk 3]
```diff
@@ -23,6 +23,7 @@
     "react-dom": "17.0.2",
     "react-i18next": "11.16.8",
     "react-router-dom": "6.3.0",
+    "react-select": "^5.7.0",
```
[the remark this answers is not in the document]
Comment (shonsirsha) [reply frontends/web/package.json]: Great, thanks for letting me know!

**Comment by shonsirsha** [comment]
Changes made, thanks for the review! @thisconnect

File: frontends/web/src/routes/buy/components/countryselect.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+const SelectSingleValue: FunctionComponent<SingleValueProps<TOption>> = (props) => {
+  const { label, value } = props.data;
+  return (
+    <div className={styles.singleValueContainer}>
+      <SelectedRegionIcon regionCode={value.toLowerCase()} />
+      <components.SingleValue {...props}>
+        <span className={styles.selectLabelText}>{label}</span>
+      </components.SingleValue>
+    </div>
+  );
```
Comment (thisconnect) [inline frontends/web/src/routes/buy/components/countryselect.tsx:50]: interesting  that you can also wrap components...
hope they never change their api 🤞

**Review by thisconnect — APPROVED** [review]
tested LGTM with small nit about TS (`TOption`)

tested:
- that country dropdown has flags
- qt app macOS
- user can select no country
- some country is guessed if missing in config.json
- app takes country from config if there is any

**Pull request merged**